core/config/seatConfig.js

```diff
@@ -0,0 +1,54 @@
+import harden from '@agoric/harden';
+
+import { makeCoreMintKeeper } from './coreMintKeeper';
+import { seatStrategy } from './strategies/seatStrategy';
+
+/**
+ * `makeSeatConfigMaker` exists in order to pass in two makeUseObj
+ * functions, one for payments and one for purses. A "use object" has
+ * all of the non-ERTP methods for assets that are designed to be
+ * used. For instance, a stock might have vote() and
+ * claimCashDividends() as methods. The use object is associated with
+ * an underlying asset that provides the authority to use it.
+ * @param {function} makeUseObjForPayment creates a "use object" for
+ * payments
+ * @param {function} makeUseObjForPurse creates a "use object" for
+ * purses
+ */
+function makeSeatConfigMaker(makeUseObjForPayment, makeUseObjForPurse) {
+  function makeSeatConfig() {
+    return harden({
+      makeCustomPayment(superPayment, issuer) {
+        const payment = harden({
+          ...superPayment,
+          // This creates a new use object which destroys the payment
+          unwrap: () => makeUseObjForPayment(issuer, payment),
```

Comment:
Why was the switch from makePTrait to makeCustomP helpful? It looks like the traits also had access to the issuer and payment when they were necessary. Couldn't the payment have been destroyed in that framework?

The improvement isn't obvious, and it has the drawback of allowing overriding of methods from the core set.

Reply:
I think there's a misunderstanding here. The `super` or `core` object was passed in for `makePTrait`, but the custom code there had no access to the full object (super/core + custom methods). By moving to `makeCustomP`, the custom code does have access to the full object, because they are creating it and returning it.

Happy to talk about this more, but to be clear, this wasn't an "improvement" - it was necessary to be able to burn the underlying payment in the custom code.